refactor: prepare Vue 3 migration

susnux opened the pull request:
* kick start https://github.com/nextcloud/server/issues/55428

## Summary

1. move legacy building to `build/frontend-legacy` (vue 2 apps)
  - apps are symlinked to be builded with webpack as before
2. build script in root is now for Vue 3 apps
3. migrated `sharebymail` as an example app.

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)
- [x] [Labels added](https://github.com/nextcloud/server/labels) where applicable (ex: bug/enhancement, `3. to review`, feature component)
- [x] [Milestone added](https://github.com/nextcloud/server/milestones) for target branch/version (ex: 32.x for `stable32`)

susnux committed:
chore(Template): allow styles from `dist` folder

Signed-off-by: Ferdinand Thiessen <opensource@fthiessen.de>

szaimen reviewed (approved):
🐘

artonge commented on .github/workflows/node-test.yml:
What are the `-ws` flags?

artonge reviewed (commented):
The build files from both systems are ending up in the `dist` directory? Could there be some conflicts?

susnux committed:
chore: split frontend building into legacy Vue 2 and Vue 3

- Built the frontend in separate packages until we migrated everything
  to Vue 3.
- Separate logic into two packages controlled by main package.json

Signed-off-by: Ferdinand Thiessen <opensource@fthiessen.de>

susnux committed:
chore: move icons building from core sources to build helpers

Signed-off-by: Ferdinand Thiessen <opensource@fthiessen.de>

susnux committed:
chore: adjust linter configs for new layout

Signed-off-by: Ferdinand Thiessen <opensource@fthiessen.de>

susnux committed:
refactor(sharebymail): migrate app to Vue 3

Signed-off-by: Ferdinand Thiessen <opensource@fthiessen.de>

susnux committed:
chore: compile assets

Signed-off-by: Ferdinand Thiessen <opensource@fthiessen.de>

susnux commented:
Basically nothing changed compared to before except that I split all complex side quests into other PRs that are now merged:
- Migrate to `@nextcloud/dialogs` v7: #55726 
- Migrate to `@nextcloud/password-confirmation` v6: #55377
- Migrate to `@nextcloud/e2e-test-server`: #55885 

So this now only does a few things:
- allow styles also from `dist` similar as with js assets
- legacy apps are built in `build/frontend-legacy` with their dependencies
- migrated apps are also built in `build/frontend` but the dependencies are already kept in the root for simpler switch once everything is migrated
- Migrated first app (sharebymail) to Vue 3

How to migrate apps?
1. Migrate the code to Vue 3
2. Remove symlink from `build/frontend-legacy/apps`
3. Remove entry from `build/frontend-legacy/webpack.modules.cjs`
4. Add symlink to `build/frontend/apps`
5. Add entry to `build/frontend/vite.config.ts`
